src/tiledb/cloud/bioimg/ingestion.py

```diff
@@ -125,6 +132,15 @@ def ingest_tiff_udf(
         from tiledb.bioimg import Converters
         from tiledb.bioimg import from_bioimg
 
+        converter = kwargs.get("converter", None)
+        user_converter = Converters.OMETIFF
+        if not converter or converter == "tiff":
+            user_converter = Converters.OMETIFF
+        elif converter == "zarr":
+            user_converter = Converters.OMEZARR
+        elif converter == "osd":
+            user_converter = Converters.OSD
```

Comment:
This could be combined with the check below so it all happens here, in one place, something like

```python
# earlier, at the module level
_SUPPORTED_CONVERTERS = {
    None: Converters.OMETIFF,  # default
    "tiff": Converters.OMETIFF,
    ...
}

# wherever the check ends up
converter_name = kwargs.get("converter")
try:
    converter = _SUPPORTED_CONVERTERS[converter_name]
except KeyError as ke:
    raise ValueError(
        f"{converter_name!r} is not a supported converter;"
        f" please select one of {set(_SUPPORTED_CONVERTERS)}"
    )
```

A couple other small things to note in the above:

- when doing `some_dict.get(...)`, None is the default default so it doesn’t need to be explicitly included
- Instead of using backslashes in strings, you can use the fact that adjacent string literals are concatenated to get nice formatting without weird syntax.

I would also recommend putting this conversion at the outer function level, so that the user gets this feedback *immediately*, before the graph is built and starts trying to be executed.

Reply:
I agree with your comments. However the reason I approached it this way is because the Coverters enum is not visible to the Cloud-Py. The Enum is defined in the tiledb-bioimg package and this is why I needed to pass them through the UDF without using them directly. The UDF on the other hand has access to the tiledb.bioimg where the converters are visible, hence the check lies inside the UDF. Is there any alternative which I can't see?

Reply:
I see! That makes sense. Might be worth adding a small comment about this (something like "we might not have tiledb.bioimg installed so we can't directly use the Converters enum here").